File: src/SimuladorDeObjetos.BusinessLogic/Entities/ClassModel.cs
using System;
using System.Collections.Generic;

namespace Domain.Entities;

    public class ClassModel
    {
        public Guid Id { get; set; }
        public string? Name { get; set; }
        public bool IsAbstract { get; set; }
        public Guid? BaseClassId { get; set; }
        public bool IsSealed { get; set; }

        public List<AttributeModel> Attributes { get; set; } = new List<AttributeModel>();
        public List<MethodModel> Methods { get; set; } = new List<MethodModel>();
    }

File: simulador-objetos-ui/src/app/models/class.model.ts
import { AttributeModel } from './attribute.model';
import { MethodModel } from './method.model';
import { InterfaceModel } from './interface.model';

export interface ClassModel {
  id: number;
  name: string;
  namespaceId: number;
  attributes: AttributeModel[];
  methods: MethodModel[];
  implementedInterfaces: InterfaceModel[];
  isStatic: boolean;
  baseClassId?: number;
  baseClass?: ClassModel;
} 